### PR TITLE
review again library search

### PR DESCRIPTION
**Pull request by remicollet**

### CODE REVIEW

**Comment by remicollet** [comment]
> Do you know if this works when installing as root?

I don't see any reason (despite I never run any build as root)

**Comment by rtheunissen** [comment]
@remicollet  build failing in Travis. I think we should also remove the `locate` call if we don't need it. I would rather use the expected, standard lib paths.

**Comment by remicollet** [comment]
travis seems happy

again hit by strange debian path
I'm not very happy to have to define a debian specific case... :(
Really will be nice if the library provide some reliable pkg-config file

File: config.m4 [hunk 1]
```diff
@@ -1,29 +1,53 @@
...
+        break
+      elif test -r $i/lib/libmpdec.$SHLIB_SUFFIX_NAME || test -r $i/lib/libmpdec.a; then
+        LIBMPDEC_DIR=$i/lib
         AC_MSG_RESULT(found in $LIBMPDEC_DIR)
         break
       fi
     done
 
-    if test -z "$LIBMPDEC_DIR"; then
+    if test -z "$LIBMPDEC_DIR" && test -r /usr/lib/x86_64-linux-gnu/libmpdec.so ; then
```
Comment (rtheunissen) [inline config.m4]: This should be symlinked under /usr/, I don't think we should need this? Not sure though. Looking at how other extensions are doing it seems a lot more simple. Is libmpdec doing something non-standard here?

Comment (remicollet) [reply config.m4]: No, lot of other extensions does something similar, see ext/gmp/config.m4

With last commit, this fix the non-x86_64 arch.

Comment (remicollet) [reply config.m4]: And again, proper way is pkg-config, but libmpdec don't support it...

**Comment by remicollet** [comment]
On fedora:

```
checking whether to enable decimal support... yes, shared
checking for libmpdec custom path... no
checking for libmpdec library in default path... found in /usr/lib64
checking for libmpdec headers in default path... found in /usr/include
checking for mpd_version in -lmpdec... yes

```
On debian (travis)

```
checking whether to enable decimal support... yes, shared
checking for libmpdec custom path... no
checking for libmpdec library in default path... found in /usr/lib/x86_64-linux-gnu
checking for libmpdec headers in default path... found in /usr/include/x86_64-linux-gnu
checking for mpd_version in -lmpdec... yes
```

**Comment by remicollet** [comment]
> Looking at how other extensions are doing it seems a lot more simple.

Lot of extensions only check for headers location, and often headers are in /usr/include, as only arch specific ones have to be in some arch specific dir, but libmpdec have a single header.

Also /usr/lib/<arch> is in default build path, so doesn't really need to be added.

File: config.m4 (hunk 1 above)
Comment (rtheunissen) [reply config.m4]: I'll email the author about it. :+1:

**Comment by rtheunissen** [comment]
LGTM :sparkles:

**Pull request merged**